hydra-node/src/Hydra/API/Server.hs

```diff
@@ -37,11 +45,17 @@ withAPIServer ::
   (ClientInput tx -> IO ()) ->
   ((ServerOutput tx -> IO ()) -> IO ()) ->
   IO ()
-withAPIServer host port tracer requests continuation = do
+withAPIServer host port tracer inputHandler continuation = do
   responseChannel <- newBroadcastTChanIO
-  let sendOutput = atomically . writeTChan responseChannel
+  history <- newTVarIO []
+  let sendOutput output = atomically $ do
+        modifyTVar' history (Right output :)
+        writeTChan responseChannel output
+  let recvInput input = do
+        atomically $ modifyTVar' history (Left input :)
+        inputHandler input
```

Comment:
Is this also echoing the inputs? I yes, I would have expected to only get the outputs latched.. both is fine right now, but I imagine a proper client having problems with this (as it expects the potentially different output data structure on the connection).

Reply:
Yes indeed, I thought it was the original intent. If we decide otherwise it's easily removed though.